packages/compiler-sfc/src/compileScript.ts

```diff
@@ -873,7 +873,8 @@ export function compileScript(
         destructured.default.end!
       )
       const isLiteral = destructured.default.type.endsWith('Literal')
-      return isLiteral ? value : `() => (${value})`
+      const isFunction = destructured.default.type.includes('Function')
+      return isLiteral || isFunction ? value : `() => (${value})`
```

Comment:
I don't it's necessary to add `() =>` for the default value. I think it should be handled by the user self.

Reply:
> I don't it's necessary to add `() =>` into the default value. I think it should be handled by the user self.

Yes, returning ()=> behaves inconsistently with other syntax such as option api, withDefault(), composition api, but this seems intentional, and removing ()=> may would be a breaking change, so I think this needs further discussion